src/index.js

```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import App from './app';
+
+ReactDOM.render(<App />, document.getElementById('app'));
+
+if (module.hot) {
+  module.hot.accept('./app', () => {
+    const NextApp = require('./app').default;
```

Comment:
@avpaul can this be converted to es6

Reply:
I don't think so! This is different from `module.export`. But if you have a suggestion, I would like to see it!

Reply:
I am also curious why do we not do something like
```javascript
const NextApp = App // Which is defined/imported above
```

Reply:
Thanks @me-x-mi, that can also work. You have to import the module when you don't have it already imported.